variables.tf

```diff
@@ -0,0 +1,172 @@
+/**
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "project_id" {
+  description = "The project ID to host the cluster in (required)"
+}
+
+variable "name" {
+  description = "The name of the cluster (required)"
+}
+
+variable "description" {
+  description = "The description of the cluster"
+  default     = ""
+}
+
+variable "regional" {
+  description = "Whether is a regional cluster (zonal cluster if set false. WARNING: changing this after cluster creation is destructive!)"
+  default     = true
+}
+
+variable "region" {
+  description = "The region to host the cluster in (required)"
+}
+
+variable "zone" {
+  description = "The zone to host the cluster in (required if is a zonal cluster)"
+  default     = ""
+}
+
+variable "additional_zones" {
+  type        = "list"
+  description = "The zone to host the cluster in (optional: only used if is a zonal cluster)"
```

Comment:
Appreciate the work you guys are doing and hope you don't mind if I add an unsolicited comment.

I don't believe this is true:

> only used if a zonal cluster

It's been my experience that `additional_zones` is treated like the gke cluster '[locations](https://cloud.google.com/kubernetes-engine/docs/reference/rest/v1beta1/projects.zones.clusters/locations)' attribute by the Google Terraform provider, for both regional and zonal clusters, as both can optionally specify a list of "locations" for nodes to be created.  The only difference is that the list of locations for zonal clusters includes the sum of `zone` and `additional_zones`.  To address this discrepancy, while allowing both zonal and regional clusters to be created in the same module, I ended up with a `zones` variable, where a user can specify a list of zones, with the first element being used for the `zone` attribute with zonal clusters.
e.g.,: 
Zonal:
`zone = "${var.zones[0])}"`
`additional_zones = "${slice(var.zones,1,length(var.zones))}"`

Regional:
`additional_zones = "${var.zones}"`

Reply:
@qvallance You are 100% correct, thanks for pointing it out.

@ryanckoch Can we please update this to take a "zones" parameter which matches @qvallance's logic above? For zonal clusters it should use the first zone is the primary. For regional, all the zones would be passed in.

Reply:
@qvallance Yeah in the current context of how we have the module built out, we split zonal and regional into to separate resource w/ a count and only currently passing additional_zones to zonal_primary. I definitely like the above idea and thanks for the feedback!

@morgante Yeah I'll get that functionality added in.

Reply:
This is now implemented.